custom/nic_compliance/static/nic_compliance/js/encoder.js

```diff
@@ -0,0 +1,53 @@
+hqDefine('nic_compliance/js/encoder.js', function () {
+    function HexParsr() {
+        function paddingStr() {
+            var s = Math.random().toString(36).slice(2, 8);
+            return s.length === 6 ? s : paddingStr();
+        }
+
+        // private property
+        var _paddingLeft = "sha256$" + paddingStr();
+        var _paddingRight = paddingStr() + "=";
+
+        var b64EncodeUnicode = function(str){
+            return encodeURIComponent(str).replace(/%([0-9A-F]{2})/g, function(match, p1) {
+                return String.fromCharCode('0x' + p1);
+            });
+        };
+
+        this.addPadding = function(secret_password) {
+            return _paddingLeft + secret_password + _paddingRight;
+        };
+
+        this.encode = function(password) {
+            password = password.trim();
```

Comment:
> Why are we trimming the password?

Since we are obfuscating the password now, even a blank string gets encoded to a not-empty string and submits to server whereas otherwise if obfuscating is not enabled the browser does not allow submit. So just added the trim.

I tried submitting with a password and then the same password along with an empty space in beginning, where both got successfully submitted. So i thought this was safe here.
If it can be dangerous, i can remove it. It would just allow someone to enter blank passwords in case obfuscation is enabled but would eventually get a message that password can not be empty from server instead of proactively from browser.